pkg/csi/service/wcp/controller.go

```diff
@@ -499,7 +500,7 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 	}
 	candidateDatastores := append(sharedDatastores, vsanDirectDatastores...)
```

Comment:
We can do that but I don't think it will offer any added benefit. 

This is because, in this example, candidate datastore list is being fetched here. But we also look at other datastores inside `CreateBlockVolumeUtil` and add them to the createvolume spec if they're compatible. For example we have a check for datastoreURL inside the storageclass.

As we're coming up with more candidate datastores inside the function itself, I feel it'll be better to filter them all out inside the function.

Reply:
Okay makes sense.